src/main/java/net/rubygrapefruit/platform/internal/jni/OsxFileEventFunctions.java

```diff
@@ -16,10 +16,51 @@
 
 package net.rubygrapefruit.platform.internal.jni;
 
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.NativeIntegration;
+import net.rubygrapefruit.platform.file.FileWatcher;
+import net.rubygrapefruit.platform.file.FileWatcherCallback;
 import net.rubygrapefruit.platform.internal.FunctionResult;
 
-public class OsxFileEventFunctions {
-    public static native void createWatch(String path, FunctionResult result);
-    public static native void startWatch(DefaultOsxFileEventFunctions.ChangeCallback callback, FunctionResult result);
-    public static native void stopWatch(FunctionResult result);
+import java.util.Collection;
+
+public class OsxFileEventFunctions implements NativeIntegration {
+
+    public FileWatcher startWatch(Collection<String> paths, double latency, FileWatcherCallback callback) {
+        if (paths.isEmpty()) {
+            return FileWatcher.EMPTY;
+        }
+        FunctionResult result = new FunctionResult();
+        FileWatcher watch = OsxFileEventFunctions.startWatch(paths.toArray(new String[0]), latency, callback, result);
+        if (result.isFailed()) {
+            throw new NativeException("Failed to start watch. Reason: " + result.getMessage());
+        }
+        return watch;
+    }
+
+    private static native FileWatcher startWatch(String[] path, double latency, FileWatcherCallback callback, FunctionResult result);
+    private static native void stopWatch(Object details, FunctionResult result);
+
+    // Created from native code
+    @SuppressWarnings("unused")
+    private static class WatchImpl implements FileWatcher {
+        private Object details;
```

Comment:
Can you add some doc to the `details` object what it is used for? I took me some time to understand. Declaring the type to `java.nio.ByteBuffer` would help as well.

Reply:
The `details` are just whatever the native code wants to keep track of. I will add docs.

I want to keep this as an opaque `Object`, though. The Java side has no business with the contents, so it shouldn't have any knowledge about it. `Object` captures that best I think, at least better than `ByteBuffer`.

Reply:
FTR, I want to reuse this type for Windows as well.

Reply:
👍